### PR TITLE
Add a Lambda+Codepipeline Typescript example

### LINKED ISSUE #267: Lambda+CodePipeline example
### General Information

* [X] :wave: I may be able to implement this feature request

* [ ] :warning: This feature might incur a breaking change 

### Description
Although there are plenty of examples in this repo that deploy a Lambda+ApiGateway stack, none of them cover the intricacies of moving the setup to a working CI. Now that Full CI/CD is becoming a well-adopted best practices across teams worldwide, not having a CDK example with a CI would impact teams looking to adopt it.

### Proposed Solution
We can leverage CodeSuite (CodeCommit, CodeBuild, and CodePipeline specifically) to cover #1 how to create a CI with CDK #2 what setup is required to link that CI to a Lambda function created with CDK as well

### Environment

  - **CDK CLI Version:** 1.31.0 (build 8f3ac79)
  - **Example:** `lambda-api-ci`
  - **Example Version:** 1.0
  - **OS:** OSX High Sierra
  - **Language:** Typescript

### PR DESCRIPTION
**Pull request by gustakasn0v**
I'm adding an example here that deploys a Lambda+ApiGateway setup using CodePipeline. 

Although there's plenty of examples of Lambda already, deploying them via CodePipeline had some non-trivial challenges to sort out. This could mean teams start using CDK and like it, but drop it because they can't adopt Full CI/CD easily when using it

Fixes #267 

---

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license].

[Apache 2.0 license]: https://www.apache.org/licenses/LICENSE-2.0

### CODE REVIEW

**Comment by gres147679** [comment]
To the reviewer: regarding Conventional Commits, would you prefer if I squashed all the commits to something like:

> feat: Add Lambda+CodePipeline Typescript example

File: typescript/lambda-api-ci/lib/ci-stack.ts [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+          service:"lambda",
+          resource: "function",
+          sep: ":",
+          resourceName: lambdaFunctionName},
+        ),
+        "arn:aws:s3:::cdktoolkit-stagingbucket-*"
+    )
+    const editOrCreateLambdaDependencies = new PolicyStatement();
+    editOrCreateLambdaDependencies.addActions(
+"iam:GetRole",
```
Comment (NGL321) [inline typescript/lambda-api-ci/lib/ci-stack.ts:89]:
```suggestion
        "iam:GetRole",
```

File: typescript/lambda-api-ci/lib/lambda-api-stack.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    this.restApi = new RestApi(this, "RestApi", {
+      deployOptions: {
+        stageName: "beta",
+        metricsEnabled: true,
+        loggingLevel: MethodLoggingLevel.INFO,
+        dataTraceEnabled: true
+      }
+    })
+
+    const lambdaPolicy = new PolicyStatement();
```
Comment (NGL321) [inline typescript/lambda-api-ci/lib/lambda-api-stack.ts:35]: Please be consistent with semicolons. We recommend all lines terminate with semicolon, but if you want to follow the no-semicolon TS guidance, stick to that throughout the example.

**Comment by gustakasn0v** [comment]
@NGL321 Thanks for taking a look! I'll address that and update the pull request

**Comment by gustakasn0v** [comment]
@NGL321 The build seems to be failing with the following error:
```
> widgets-lambda@1.0.0 build /Users/seoaneg/Workspaces/aws-cdk-examples/typescript/lambda-api-ci/cdk.out/asset.5d8ebe4e1b7da78e3a974de62e8d9aac8a42747f5c303e249b6d1e2855c6e92f
> tsc

npx: installed 202 in 17.326s
--app is required either in command-line, in cdk.json or in ~/.cdk.json
```

`widgets-lambda` is the name of the NPM package I created under `typescript/lambda-api-ci/src` for the Lambda function. It seems the build is mistaking it for a CDK package. Do you know if I can exclude it somehow?

**Comment by gustakasn0v** [comment]
Screen Shot 2020-04-18 at 5 50 34 pm

File: scripts/build-npm.sh [hunk 3]
```diff
@@ -20,7 +20,7 @@ verify_star_dependencies() {
 }
 
 # Find and build all NPM projects
-for pkgJson in $(find typescript -name package.json | grep -v node_modules); do
+for pkgJson in $(find typescript -name cdk.json | grep -v node_modules); do
```
Comment (gustakasn0v) [inline scripts/build-npm.sh:23]: I've run the following commands to compare the directories that are evaluated:
```
find typescript -name package.json  | grep -v node_modules | xargs -n 1 dirname | sort
find typescript -name cdj.json  | grep -v node_modules | xargs -n 1 dirname | sort
```

Here is the diff. Directories removed are shown in grey on the left-hand side:
Screen Shot 2020-04-18 at 5 50 34 pm
[Link](https://user-images.githubusercontent.com/2453371/79673590-02022900-81d3-11ea-9f22-d7eb87153230.png)

Comment (NGL321) [reply scripts/build-npm.sh:23]: Thank you for the catch! I had not noticed this. The fix is logical and LGTM.

😸

**Comment by gustakasn0v** [comment]
@NGL321 Looks like mergify discarded your approval after the rebase. Can you take another look?

**Comment by NGL321** [comment]
> @NGL321 Looks like mergify discarded your approval after the rebase. Can you take another look?

Well thats frustrating! Taking care of now!

😸 😷

**Comment by gustakasn0v** [comment]
@NGL321 I think it blew up again 🤦

**Pull request merged**